### PR TITLE
Mainline sleep deprivation

### PR DESCRIPTION
**Pull request by anothersimulacrum**
#### Summary
SUMMARY: Mods "Mainline sleep deprivation"

#### Purpose of change
No reason for this to not be in the base game

#### Describe the solution
Move everything from sleep deprivation to the base game, mark the mod as obsolete.

#### Testing
Tests pass (I got failures, but they also occurred on master (or not, I looked at the wrong tests window)).
Waiting for several days without sleeping gives me sleep deprivation (using atomic coffee when I get too tired).
Sleeping eliminates the negative effects of sleep deprivation.

#### Additional context
The CBM may need some review, I think the rest is probably fine.
I need to fix hunger tests.

### CODE REVIEW

**Comment by SirPendrak** [comment]
For some reason when sleep deprivation was only a mod, magnesium tablets allways spawned in stacks of **one**. I can figure it out why, by looking at the code, or if it is fixed now.

**Comment by I-am-Erk** [comment]
I don't really love magnesium being used like this... Magnesium is sometimes given for sleep problems because deficiency can cause less restful sleep.

You could just rename it Melatonin.

**Comment by I-am-Erk** [comment]
I won't content freeze this because I think it's a reasonable thing to get in before 0.E. However I'd really like to see magnesium name-changed, if you don't disagree.

**Comment by anothersimulacrum** [comment]
I don't, just a little busy and trying to fix test failures first.

**Comment by anothersimulacrum** [comment]
Well that was an extremely annoying test failure. Should be resolved now.
Magnesium has been renamed to melatonin.

File: src/iuse.cpp [hunk 1]
```diff
@@ -9693,14 +9693,14 @@ int iuse::gobag_personal( player *p, item *it, bool, const tripoint & )
     return gobag( p, it, true );
 }
 
-int iuse::magnesium_tablet( player *p, item *it, bool, const tripoint & )
+int iuse::melatonin_tablet( player *p, item *it, bool, const tripoint & )
```
Comment (ymber) [inline src/iuse.cpp]: Why do we need an iuse for this?

File: data/json/game_balance.json [hunk 2]
```diff
@@ -130,7 +130,7 @@
     "name": "SLEEP_DEPRIVATION",
     "info": "Enables sleep deprivation mechanics. If true, stimulants will only help you so far, and you might have to enforce proper sleep hygiene for a while.",
     "stype": "bool",
-    "value": false
+    "value": true
```
Comment (ymber) [inline data/json/game_balance.json]: This option should be removed now.

File: src/iuse.cpp (hunk 1 above)
Comment (anothersimulacrum) [reply src/iuse.cpp]: Presumably for the message if you've already taken it. Not a super strong justification, but that's what I can come up with.

Comment (ymber) [reply src/iuse.cpp]: I think that's information we shouldn't give to the player. We don't warn for excessive use of other drugs.

Comment (anothersimulacrum) [reply src/iuse.cpp]: Alright, will look into cutting that.

Comment (I-am-Erk) [reply src/iuse.cpp]: It's worth noting that additional doses of melatonin just generally don't do anything

**Pull request merged**